### DISCUSSION TITLE
Is it right way to use useWatch hook conditionally ?

### DISCUSSION THREAD

**Discussion by abdulmateentechbits** [react-hook-form/react-hook-form#9411] [General]
Here I am using **useWatch** hook conditionally like bellow for an array value (cr_heads)

`const crHeadArray = useWatch({ control, name: 'cr_heads' }) || [];`

Why I am using it conditionally?

It's because of when **useWatch** can not watch that value has change it return undefined it cause and error when I use forEach loop over crHeadArray.
That's why I am using **useWatch** conditional if **useWatch** can not see any changes rather then return undefined it will return empty array so there were no issue in forEach loop

**Comment by bluebill1049** [comment]
no, that's a rule of react custom hook. you can use the `disbaled` prop to pause the subscription.
